bmm/src/main/java/org/openehr/bmm/persistence/validation/BmmMessageIds.java

```diff
@@ -6,85 +6,108 @@
+    ec_BMM_GPGPT(I18n.register("Schema {0} class definition {1} generic property {2} generic parameter {3} not found in schema or in containing class declarations (if open)")),
+    ec_BMM_GPGPM(I18n.register("Schema {0} class definition {1} marked ''is_generic'' but has no generic parameter declarations")),
+    ec_BMM_GPGPU(I18n.register("Schema {0} class definition {1} generic property {2} type {3} parameter {4} not found in class definitions or {3} formal declaration")),
+    ec_BMM_SPT(I18n.register("Schema {0} class definition {1} single-valued property {2} type {3} not found in schema")),
+    ec_BMM_SPOT(I18n.register("Schema {0} class definition {1} single-valued property {2} open generic parameter {3} not found in containing class declarations")),
+    ec_BMM_PKGCL(I18n.register("Schema {0} class {1} mentioned in package {2} but not in schema, or relevant child schema")),
+    ec_BMM_PKGID(I18n.register("Schema {0} class definition {1} not declared in any package")),
+    ec_BMM_PKGTL(I18n.register("Schema {0} top-level sibling package definitions cannot include a package which is the child of another")),
+    ec_BMM_PKGQN(I18n.register("Schema {0} packages with qualified name found in package {1} (qualified names not allowed except at top-level)")),
+    ec_BMM_PKGCE(I18n.register("Schema {0} package {1} contains class with empty name")),
+    ec_BMM_CLPKDP(I18n.register("Schema {0} has duplicate class name {1} in package {2} and also package {3}")),
+    ec_BMM_CLDUP(I18n.register("Schema {0} has duplicate class name {1} in class definitions")),
+    ec_BMM_MDLPK(I18n.register("Schema {0} archetype_closure_package {1} does not exist")),
+    ec_BMM_PRDUP(I18n.register("Schema {0} class {1} duplicate property within class {2}")),
+    ec_BMM_PRNCF(I18n.register("Schema {0} class {1} property {2} does not conform to same property in ancestor {3} (duplicate?)")),
+    ec_BMM_INCERR(I18n.register("Schema {0} included schema {1} validity failure")),
+    ec_BMM_INCWARN(I18n.register("Schema {0} included schema {1} validity warning")),
+    ec_BMM_ARPAR(I18n.register("Schema {0} archetype parent class {1} not defined in schema")),
+    ec_BMM_RMREL(I18n.register("Schema {0} RM release {1} not valid; should be 3-part numeric version")),
     //Added for java-model-stack
-    public static final String ec_object_file_not_valid = "object_file_not_valid";
-    public static final String ec_bmm_schema_load_error = "bmm_schema_load_failed";
+    ec_object_file_not_valid(I18n.register("EC Object file not valid")),
+    ec_bmm_schema_load_error(I18n.register("BMM Schema load error"));
+
+    private final String template;
+
+    BmmMessageIds(String template) {
+        this.template = template;
+    }
+
+    @Override
+    public String getCode() {
+        String code = name();
+        if(code.startsWith("ec_")) {//it''s possible to get rid of the ec_ prefix altogether
```

Comment:
yes, they should be. I'm not sure why the prefix is there - it's not my code and I left it in there  because there could be valid reasons for a differentiation between error codes and other things.